src/allotropy/parsers/agilent_gen5/agilent_gen5_structure.py

```diff
@@ -395,6 +403,17 @@ def _get_filter_sets(
                 mirror=mirror,
                 optics=optics[idx] if optics else None,
             )
+        if read_mode == ReadMode.FLUORESCENCE:
+            for label in measurement_labels:
```

Comment:
Note for reviewer: Since these measurement labels were attached to the way we create the filter sets, what I did is to duplicate the filter having the extra label pointing to the new one. This way we can support for multiple filter sets. Although not a very good looking solution, I appreciate any suggestion on the approach.

Reply:
I agree with your approach to duplicate the measurement labels that may hve multiple formats.

However, as you note, there is some code smell here. My main concern if the line above, (`if idx >= len(gains)`...). It shows that we are abandoning a condition that we knew to be true to do this new approach.

I suggest this alternative: when creating measurement labels, store the extra labels as a separate structure - a mapping from measurement label to extra aliases that label might have.

Then, in this function, you can (a) match gains to the original measurement label, and (b) you can use the "extra labels" structure to create the duplicated filter sets. 

Then, when storing the measurement labels in the main create function, you can get the same result by doing measurement_labels = measurement_labels + all extra labels.